src/lib/pages/proposal/constants.tsx

```diff
@@ -1,115 +1,112 @@
+          to an allowed list.
+        </span>
+      ),
+    },
+  }[permission]);
 
 const storeCodePage = "proposal-store-code";
-export const SIDEBAR_STORE_CODE_DETAILS: SidebarDetails = {
-  mainnet: {
-    page: storeCodePage,
-    title: "Why do I need to submit proposal?",
-    description: (
-      <span>
-        Osmosis Mainnet is permissioned chain, which means you will need to
-        submit proposal to store code
-        <br />
-        <br />
-        Another way is to get your wallet address to allow list to store code
-        without opening proposal.
-        <br />
-        <br />
-        You still can upload your Wasm to see how your code works on Testnet.
-      </span>
-    ),
-    toNetwork: true,
-  },
-  testnet: {
-    page: storeCodePage,
-    title: "Do I need to submit Proposal to store code in testnet?",
-    description: (
-      <span>
-        On the Osmosis testnet, you can store code without submitting a proposal
-        by using
-        <Link
-          href="/deploy"
-          onClick={() =>
-            AmpTrackUseRightHelperPanel(storeCodePage, "to-/deploy")
-          }
-        >
-          {" "}
-          <Text as="span" color="secondary.main">
-            Deploy Contract
-          </Text>
-        </Link>
-        <br />
-        <br />
-        However, if you&apos;re interested in exploring about the proposal
-        process, you can try submitting a proposal.
-        <br />
-        <br />
-        On the mainnet, submitting a proposal is required to store code.
-      </span>
-    ),
-    toPage: true,
-    toPagePath: "/deploy",
-    toPageTitle: "Go To Deploy Contract",
-    toNetwork: true,
-  },
-  // TODO: fill localnet information
-  localnet: {
-    page: storeCodePage,
-    title: "",
-    description: <span />,
-  },
-};
+export const SIDEBAR_STORE_CODE_DETAILS: (
+  chainName: string,
+  permission: NetworkPermission
+) => SidebarMetadata = (chainName, permission) =>
+  ({
+    permissioned: {
+      page: storeCodePage,
+      title: "Why do I need to submit proposal?",
+      description: (
+        <span>
+          {chainName} is permissioned chain, which means you will need to submit
+          proposal to store code
+          <br />
+          <br />
+          Another way is to get your wallet address to allow list to store code
+          without opening proposal.
+          <br />
+          <br />
+          You still can upload your Wasm to see how your code works on Testnet.
+        </span>
+      ),
+    },
+    permissionless: {
+      page: storeCodePage,
+      title: "Do I need to submit Proposal to store code in testnet?",
+      description: (
+        <span>
+          On the {chainName}, you can store code without submitting a proposal
+          by using
+          <Link
+            href="/deploy"
+            onClick={() =>
+              AmpTrackUseRightHelperPanel(storeCodePage, "to-/deploy")
```

Comment:
@poomthiti I'm not sure about this why use `to-/deploy`?

Reply:
maybe trying to follow a kebab case, like to-[pathname]